Bottom: e67d88d17cb26ea5305c4b691d21100a48de6508
Top:    384e5e94782f25233af9c7a4b748775beb5e4fca
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-08-02 13:35:53 +0800

Refresh of tuntap-flush

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index f9d3d29..7951012 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1626,9 +1626,10 @@ static u32 tun_do_xdp(struct tun_struct *tun,
 	switch (act) {
 	case XDP_REDIRECT:
 		*err = xdp_do_redirect(tun->dev, xdp, xdp_prog);
-		xdp_do_flush_map();
-		if (*err)
+		if (*err) {
+			act = XDP_DROP;
 			break;
+		}
 		goto out;
 	case XDP_TX:
 		*err = tun_xdp_tx(tun->dev, xdp);
